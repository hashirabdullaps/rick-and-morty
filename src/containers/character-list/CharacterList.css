.characterList {
    padding: 1em;
    background: #c09b7569;
}

.characterList h1 {
    font-size: 35px;
    text-align: center;
    font-weight: bold;
    color: #562e36;
}


.characterList_scroll {
    overflow-x: hidden !important;
}