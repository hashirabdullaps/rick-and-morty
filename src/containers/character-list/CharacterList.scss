.characterList {
    padding: 1em;
    // background: #c09b7569;
    font-family: monospace;
    background-image: url('../../assets/images/space.jpg');

    h1 {
        font-size: 35px;
        text-align: center;
        font-weight: bold;
        color: #562e36;
    }
}

.characterList_scroll {
    overflow-x: hidden !important;
}

.card-grid {
        display: flex;
        flex-wrap: wrap;
        justify-content: space-around;
}

.scroll-top {
    position: fixed;
    bottom: 3em;
    right: 2em;
    z-index: 100;
}