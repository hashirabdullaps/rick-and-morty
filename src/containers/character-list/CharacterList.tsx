import React, { useEffect } from "react";
import { Row, Col, Spin, Space } from "antd";

import CharacterCard from "../../components/character-card/CharacterCard";

import { useFetchCharactersQuery } from "../../graphql/queries/useFetchCharactersQuery";

import "./CharacterList.css";
import InfiniteScroll from "react-infinite-scroll-component";


const CharacterList = () => {
    const {
        characters,
        count,
        currentPage,
        nextPage,
        getCharacters,
        fetchMoreCharacters
    } = useFetchCharactersQuery();

    useEffect(() => {
        getCharacters({ variables: { page: 1 } });
        console.log('called ');
    }, []);


    const loadMore = () => nextPage !== null && fetchMoreCharacters(currentPage + 1);

    return (
        <div className="characterList">
            <InfiniteScroll
                className="characterList"
                dataLength={count}
                next={loadMore}
                hasMore={true}
                loader={nextPage !== null ? <Spin className="spinner-loading" size="large" /> : <></>}
            >
                <Space direction="vertical" size="middle" style={{ display: 'flex' }}>
                    <Row gutter={[16, 16]} >
                        {characters?.map((character) => {
                            return (<Col key={`${character.id} - ${character.name}`} flex='1'>
                                <CharacterCard  {...character} />
                            </Col >)
                        })}
                    </Row>
                </Space>
            </InfiniteScroll>
        </div>
    );


};

export default CharacterList;