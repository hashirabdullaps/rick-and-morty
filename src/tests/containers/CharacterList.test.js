import React from 'react';
import renderer from 'react-test-renderer';
import CharacterList from '../../containers/character-list/CharacterList';
import { MockedProvider } from '@apollo/client/testing';
import { BrowserRouter } from "react-router-dom";
import { act } from '@testing-library/react';

describe('CharacterList component', () => {

    test('should test CharacterList component', () => {

        const mocks = [];
        act(() => {
            const component = renderer.create(
                <MockedProvider mocks={mocks} >
                    <BrowserRouter>
                        <CharacterList />
                    </BrowserRouter>
                </MockedProvider>);

            let tree = component.toJSON();
            expect(tree).toMatchSnapshot();
        })

    });

})

