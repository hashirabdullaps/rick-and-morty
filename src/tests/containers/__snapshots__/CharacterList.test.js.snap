// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterList component should test CharacterList component 1`] = `null`;
